plugins/modules/content_export_library.py

```diff
@@ -0,0 +1,99 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: content_export_library
+version_added: 3.5.0
+short_description: Manage pulp3 content exports
+description:
+    - Export library content to a directory.
+author:
+    - "Jeremy Lenz (@jeremylenz)"
+options:
+  destination_server:
+    description:
+      - Destination server name; optional parameter to differentiate between exports
+    required: false
+    type: str
+  chunk_size_gb:
+    description:
+      - Split the exported content into archives no greater than the specified size in gigabytes.
+    required: false
+    type: int
+  fail_on_missing_content:
+    description:
+      - Fails if any of the repositories belonging to this organization are unexportable.
+    required: false
+    type: bool
+  incremental:
+    description:
+      - Export only the content that has changed since the last export.
+    required: false
+    type: bool
+  from_history_id:
+    description:
+      - Export history identifier used for incremental export. If not provided the most recent export history will be used.
+    required: false
+    type: str
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.organization
+'''
+
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import KatelloAnsibleModule, _flatten_entity
+
+
+class KatelloContentExportModule(KatelloAnsibleModule):
+    pass
+
+
+def main():
+    module = KatelloContentExportModule(
+        foreman_spec=dict(
+            destination_server=dict(required=False, type='str'),
+            chunk_size_gb=dict(required=False, type='int'),
+            fail_on_missing_content=dict(required=False, type='bool'),
+            from_history_id=dict(required=False, type='str'),
+        ),
+        argument_spec=dict(
+            incremental=dict(required=False, type='bool'),
+        ),
+    )
+
+    with module.api_connection():
+        module.auto_lookup_entities()
+
+        incremental = module.params['incremental']
+        endpoint = 'content_export_incrementals' if incremental else 'content_exports'
+
+        if module.params.get('from_history_id') and incremental is not True:
+            module.fail_json(msg='from_history_id is only valid for incremental exports')
```

Comment:
My intuition about the design is that it would be nice for the module not to make these assumptions, and rather expect the user to be explicit in order to limit any potential disparities between what the module will do vs. what the user understood or intended it to do. And then to have a couple very small roles which are more workflow driven and are allowed to make nice assumptions about what the user intended from minimal inputs.

Let me give an example of why this might be beneficial. Let's say I'm modifying the data I provide to an existing role or playbook, with the intent to change an incremental export to a full export, and I currently have some data like:

```yaml
  incremental: true
  some_other: "{{ stuff }}"
  some_more: "{{ other_stuff }}"
  from_history_id: "{{ saved_export_history_id }}"
```
And then I take a quick look and make the seemingly reasonable assumption that I can flip `true` to `false` and I will get a full backup.

But Ansible assumes, without telling me so, that actually I really meant `true`, and I'm disappointed _much later on_ when I discover that I'm relying on what I thought was a full export that was in fact only incremental.

OTOH if the module catches this and tells me that I'm trying to do something that doesn't make any sense, I'll realize my error right away before any damage is done (like deleting my prior incremental backups because I think I have generated a new full backup).

Another way of thinking about this is that I am looking at the modules as being sort of an API or library for role developers. So this viewpoint is aligned with having a single module for everything related to content exports, eliminating lots of duplicated code there, and then having a couple very small and easy to use roles corresponding to the different user scenarios.

Reply:
> And then I take a quick look and make the seemingly reasonable assumption that I can flip `true` to `false` and I will get a full backup.
> 
> But Ansible assumes, without telling me so, that actually I really meant `true`, and I'm disappointed _much later on_ when I discover that I'm relying on what I thought was a full export that was in fact only incremental.

Well, yeah, I did mean to not to have an (exposed) `incremental` parameter at all, so there would be no assumption that flipping it to `false` would create a full export.

> Another way of thinking about this is that I am looking at the modules as being sort of an API or library for role developers.

I strongly disagree, if a module requires a wrapping role to function, the module is bad. Modules are not APIs, they are user facing interfaces.

Reply:
> Well, yeah, I did mean to not to have an (exposed) `incremental` parameter at all, so there would be no assumption that flipping it to `false` would create a full export.

Thank you, that makes a lot more sense than how I interpreted it.